gunrock/applications/sssp/sssp_implementation.hxx

```diff
@@ -178,20 +144,62 @@ struct sssp_enactor_t : enactor_t<problem_t> {
     operators::advance::execute<operators::advance_type_t::vertex_to_vertex,
                                 operators::advance_direction_t::forward,
                                 operators::load_balance_t::merge_path>(
-        G, enactor_type::get_enactor(), shortest_path, context);
+        G, E, shortest_path, context);
 
     // Execute filter operator on the provided lambda
     operators::filter::execute<operators::filter_type_t::uniquify>(
-        G, enactor_type::get_enactor(), remove_completed_paths, context);
+        G, E, remove_completed_paths, context);
   }
 
-  sssp_enactor_t(problem_t* _problem,
-                 std::shared_ptr<cuda::multi_context_t> _context)
-      : enactor_type(_problem, _context) {}
-
-  sssp_enactor_t(const sssp_enactor_t& rhs) = delete;            // Boilerplate? Can remove?
-  sssp_enactor_t& operator=(const sssp_enactor_t& rhs) = delete; // Boilerplate? Can remove?
-};  // struct sssp_enactor_t
+};  // struct enactor_t
+
+
+// !! Helper -- This should go somewhere else -- @neoblizz, where?
```

Comment:
I think this needs to be maybe an API in the context file that takes in devices and returns the shared ptr.

Reply:
I'll leave that to you for now -- I don't completely grasp the meaning of context and how they're going to evolve as we add multiGPU stuff.

Reply:
We can discuss this more in detail with Serban as well.